queries/asciidoc/injections.scm

```diff
@@ -0,0 +1,43 @@
+((paragraph) @injection.content
+  (#set! injection.include-children)
+  (#set! injection.language "asciidoc_inline"))
+
+((line) @injection.content
+  (#set! injection.include-children)
+  (#set! injection.language "asciidoc_inline"))
+
+((section_block
+  (element_attr
+    (element_attr_marker)
+    (attr_value) @injection.language
+    (element_attr_marker))
+  (raw_block
+    (raw_block_marker)
+    (raw_block_body) @injection.content
+    (raw_block_marker)))
+  (#gsub! @injection.language "%s*,%s*(%w+)%s*" "%1")
+  (#vim-match? @injection.language "\\s*,\\s*.*\\s*"))
```

Comment:
There are two forms of syntax:

```adoc
[, cpp]
----
int a  = 10;
----
```

or 
```adoc
[source, cpp]
----
int a = 10;
----
```

Since Lua does not have the concept of optional group, vim-regex is used here.